components/automate-cli/pkg/diagnostics/option.go

```diff
@@ -0,0 +1,75 @@
+package diagnostics
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var ErrInvalidOption = errors.New("Invalid option")
+
+type Option struct {
+	Key   string
+	Value string
+}
+
+func OptionsFromStrings(options []string) ([]*Option, error) {
+	out := make([]*Option, len(options))
+	for i, o := range options {
+		opt, err := OptionFromString(o)
+		if err != nil {
+			return nil, errors.Wrapf(err, "%q is not a valid option", o)
+		}
+		out[i] = opt
+	}
+	return out, nil
+}
+
+func OptionFromString(option string) (*Option, error) {
+	parts := strings.SplitN(option, ":", 2)
+	if len(parts) == 0 {
+		return nil, ErrInvalidOption
+	}
+	return &Option{
+		Key:   parts[0],
+		Value: parts[1],
+	}, nil
+}
+
+func (o *Option) GetString(defaultVal string) string {
+	if o == nil || o.Value == "" {
+		return defaultVal
+	}
+	return o.Value
+}
+
+func (o *Option) GetInt(defaultVal int) int {
```

Comment:
[nit] I don't think the `GetInt()` and `GetString()` method names make it very clear what's happening here, which is that we want the option option value as either a string or int, if it's set, otherwise we want the default that's given. The best I can summarize that with some brevity is `ValAsIntOr` and `ValAsStrOr`, not the best, but maybe a starting point.

Reply:
Could also see `ValueOr` and `ValueStrOr`